easybuild/easyconfigs/a/AdaptiveCpp/AdaptiveCpp-23.10.0-cpeGNU-22.12-rocm-5.2.3.eb

```diff
@@ -0,0 +1,86 @@
+easyblock = 'CMakeMake'
+
+name =    'AdaptiveCpp'
+version = '23.10.0'
+versionsuffix = '-rocm-5.2.3'
+
+homepage = 'https://adaptivecpp.github.io/'
+
+whatis = [
+    "Description: AdaptiveCPP is a modern SYCL implementation targeting CPUs and GPUs, with a focus on leveraging existing toolchains such as CUDA or HIP"
+]
+
+description = """
+hipSYCL is a modern SYCL implementation targeting CPUs and
+GPUs, with a focus on leveraging existing toolchains such as CUDA or HIP.
+
+It targets any CPU via OpenMP, NVIDIA GPUs via CUDA, AMD GPUs via HIP/ROCm and 
+Intel GPUs via oneAPI Level Zero and SPIR-V (experimental). However, this 
+module is for LUMI and does not contain the NVIDIA and Intel GPU support.
+"""
+
+toolchain = {'name': 'cpeGNU', 'version': '22.12'}
+toolchainopts = {'verbose': False, 'openmp': True}
+
+#source_urls = ["https://github.com/AdaptiveCpp/AdaptiveCpp/archive/refs/tags/"]
+#sources =     ["v%(version)s.tar.gz"]
+
+sources = [{
+    'filename': '%(name)s-%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/AdaptiveCpp',
+        'repo_name': '%(name)s',
+        'commit': 'v%(version)s',
+        'keep_git_dir': True,
+    }
+}]
+
+dependencies = [
+    ('cray-python/3.9.13.1', EXTERNAL_MODULE),
+    ('rocm/5.2.3', EXTERNAL_MODULE),
```

Comment:
While single-pass compiler is indeed a very nice feature, I doubt it is particularly relevant for LUMI, where there is only one relevant target (two if we count CPU).

Reply:
Single-pass compiler is about more than just being able to target many devices from one binary.
There will likely be additional benefits in the future, such as the ability to JIT highly optimized kernels based on runtime knowledge. This will be relevant in any performance-critical scenario.
Already today there are cases where it sometimes performs better than the old compiler, and it can thus be valuable to try it, as it's a completely independent compiler. Plus, it's also just more easy to use than the old compiler for users, which might be relevant if other users apart from Gromacs might want to use that package.

Reply:
>  The generic single-pass compiler has full support for atomics across all backends

My bad. I should not have relied on [an almost one-year-old](https://adaptivecpp.github.io/hipsycl/sscp/compiler/generic-sscp/) post.

>  But the feature difference is not one-sided: For example generic supports SYCL_EXTERNAL, which the old compiler does not. What if a user wants to use that?

Ok, good point.

Still, I don't think LUMI has newer ROCm available yet :( I was just pointing out that it is not worth stalling this PR until the software stack is upgraded.

Reply:
No worries :)

> Still, I don't think LUMI has newer ROCm available yet :( I was just pointing out that it is not worth stalling this PR until the software stack is upgraded.

I agree, we should not delay this PR for an indefinite amount of time until a newer ROCm is available, which may or may not happen anytime soon. I was just pointing out that if there's a newer stack available, it should be preferred due to the additional value for users :)